### PR TITLE
implement qr code in spacewalk bridge

### PR DESCRIPTION
**Pull request by Sharqiewicz**
### What:

Implement QR Code and Button compatible with Stellar SEP-0007 in order to allow user easily create transactions needed in Spacewalk.

### How:

Add package: qrcode.react - the most popular package for generating qr codes in React space
Create function: generateQRCodeAddress to easily create SEP-0007 URI scheme messages
Divide huge TransferDialog component into smaller ones
Implement SEP-0007 in ConfirmationDialog for Spacewalk > Issue and Redeem

### Comments:

Tested with wallets:
- Lobstr (mobile) ✅ works
- Solar (Desktop) ✅ works
- Rabet (Browser Extension) 🟥 doesn't work - Rabet doesn't listen for SEP-0007
- Freighter (Browser Extension) 🟥 doesn't work - Freighter doesn't listen for SEP-0007

Closes: #395

### CODE REVIEW

**Comment by ebma** [comment]
## `yarn.lock` changes

<details open>
<summary>Click to toggle table visibility</summary>
<br/>

| Name | Status | Previous | Current |
| :- | :-: | :-: | :-: |
| `qrcode.react` | [<sub>ADDED</sub>](#) | - | 3.1.0 |

</details>

**Comment by ebma** [comment]
## `yarn.lock` changes

<details open>
<summary>Click to toggle table visibility</summary>
<br/>

| Name | Status | Previous | Current |
| :- | :-: | :-: | :-: |
| `qrcode.react` | [<sub>ADDED</sub>](#) | - | 3.1.0 |

</details>

File: src/helpers/stellar/sep0007.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+interface GenerateSEP0007URIScheme {
+  vault_stellar_account: string;
+  issue_amount: string;
+  asset_code: string;
+  asset_issuer: string;
+  issue_request_memo: string;
```
Comment (ebma) [inline src/helpers/stellar/sep0007.ts:6]: Nit: Let's continue using camelCase notation also for these fields.

File: vite.config.ts [hunk 2]
```diff
@@ -19,7 +19,7 @@ export default defineConfig({
       plugins: [
         // Enable esbuild polyfill plugins. We use the default settings.
         // see https://github.com/cyco130/esbuild-plugin-polyfill-node?tab=readme-ov-file#options
-        polyfillNode({}),
+        // polyfillNode({}),
```
Comment (ebma) [inline vite.config.ts:22]: Do we need to remove this?

**Review by ebma — CHANGES_REQUESTED** [review]
Really nice changes, good job @Sharqiewicz 🙏 

Since we learned that the browser extensions don't support SEP-07 and 1) LOBSTR doesn't have a desktop wallet, 2) Solar does not fully support SEP-07 and 3) there don't seem to be any (popular) Stellar desktop wallets that support it, I think we should remove the extra 'Sign tx' button for now and re-add it as part of a new ticket. In this follow-up task we can then make the 'Sign tx' button trigger a signing request to the wallet browser extensions. 

Also, I think we should show at least a little helper text above the QR code so the user is not totally lost. I can imagine something like: 'Scan the QR code with a Stellar wallet'.

Besides that, the QR code scanning did not work for me on Amplitude/Foucoco. I assume it's because of the theme, showing a QR code **without a border** on a dark background. I assume this can be fixed by adding a little white border around the QR code.

**Comment by ebma** [comment]
## `yarn.lock` changes

<details open>
<summary>Click to toggle table visibility</summary>
<br/>

| Name | Status | Previous | Current |
| :- | :-: | :-: | :-: |
| `qrcode.react` | [<sub>ADDED</sub>](#) | - | 3.1.0 |

</details>

**Comment by ebma** [comment]
## `yarn.lock` changes

<details open>
<summary>Click to toggle table visibility</summary>
<br/>

| Name | Status | Previous | Current |
| :- | :-: | :-: | :-: |
| `qrcode.react` | [<sub>ADDED</sub>](#) | - | 3.1.0 |

</details>

**Comment by ebma** [comment]
## `yarn.lock` changes

<details open>
<summary>Click to toggle table visibility</summary>
<br/>

| Name | Status | Previous | Current |
| :- | :-: | :-: | :-: |
| `qrcode.react` | [<sub>ADDED</sub>](#) | - | 3.1.0 |

</details>

**Review by ebma — APPROVED** [review]
Nice, can confirm it works also on Amplitude with the dark theme now 👍

**Review by prayagd — APPROVED** [review]
Works as expected, tested using Lobstr. Prefills with the vault address, amount and the memo. Really good work @Sharqiewicz

**Comment by prayagd** [comment]
@Sharqiewicz is this done, can we merge?

**Pull request merged**